File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = len(s)
    for cut in range(1, len(s)//2 + 1):
        result = ""
        i = 0
        while i < len(s):
            j = i + cut
            count = 1
            while j < len(s) and s[i:i+cut] == s[j:j+cut]:
                j += cut
                count += 1
            

            result += str(count) + s[i:i+cut] if count > 1 else s[i:i+cut]        
            i = j
            

        
        answer = min(answer, len(result))
    return answer
            
        
        
        

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
class Solution {
    public int solution(String s) {
        int answer = s.length();
        
        for (int length = 1; length < s.length() / 2 + 1; length++) {
            int i = 0;
            int result = s.length();
            while (i <= s.length() - length * 2) {                
                int press = 0;
                String target = s.substring(i, i+length);
                while (i <= s.length() - length * 2 && target.equals(s.substring(i+length, i+2*length))) {
                    i += length;
                    press++;
                }
                
                if (press > 0) {
                    result -= press * length - Integer.toString(press+1).length();
                } else {
                    i += length;
                }
            }
            answer = (int) Math.min(answer, result);
            
        }
        
        
        
        return answer;
    }
}